charts/cortex/templates/store-gateway/store-gateway-svc.yaml

```diff
@@ -1,4 +1,5 @@
 {{- if eq .Values.config.storage.engine "blocks" -}}
+{{- if not .Values.topologyAware.enabled }}
```

Comment:
Same as for the ingester. I think the service named `{{ template "cortex-store-gateway.fullname" . }}` should always be present and the headless services should be created per availability cell.